src/codegen/compile.ml

```diff
@@ -7540,6 +7616,17 @@ type scope_wrap = G.t -> G.t
 
 let unmodified : scope_wrap = fun code -> code
 
+let potential_pointer typ : bool = 
+  (* must not eliminate nested optional types as they refer to a heap object for ??null, ???null etc. *)
+  let rec can_be_pointer typ nested_optional = 
+    
+    Type.(match normalize typ with
+    | Mut t -> (can_be_pointer t nested_optional)
+    | Opt t -> (if nested_optional then true else (can_be_pointer t true))
+    | Prim (Null| Bool | Char | Nat8 | Nat16 | Int8 | Int16) | Non | Tup [] -> false
+    | _ -> true) in
+  can_be_pointer typ false
```

Comment:
```suggestion
let rec can_be_pointer typ nested_optional =     
  Type.(match normalize typ with
  | Mut t -> (can_be_pointer t nested_optional)  
  | Opt t -> (if nested_optional then true else (can_be_pointer t true))
  | Prim (Null| Bool | Char | Nat8 | Nat16 | Int8 | Int16) | Non | Tup [] -> false
  | _ -> true) 
  
let potential_pointer typ : bool = 
  (* must not eliminate nested optional types as they refer to a heap object for ??null, ???null etc. *)
  can_be_pointer typ false
```
Might be a little faster, if less clear, depending on how OCaml optimizes local functions.

Reply:
Thank you.